src/scripts/livetvsuggested.js

```diff
@@ -1,197 +1,222 @@
 
     function setScrollClasses(elem, scrollX) {
-        scrollX ? (elem.classList.add("hiddenScrollX"), layoutManager.tv && elem.classList.add("smoothScrollX"), elem.classList.add("scrollX"), elem.classList.remove("vertical-wrap")) : (elem.classList.remove("hiddenScrollX"), elem.classList.remove("smoothScrollX"), elem.classList.remove("scrollX"), elem.classList.add("vertical-wrap"))
+        if (scrollX) {
+            elem.classList.add("hiddenScrollX");
+
+            if (layoutManager.tv) {
+                elem.classList.add("smoothScrollX");
+            }
+
+            elem.classList.add("scrollX");
+            elem.classList.remove("vertical-wrap");
+        } else {
+            elem.classList.remove("hiddenScrollX");
+            elem.classList.remove("smoothScrollX");
+            elem.classList.remove("scrollX");
+            elem.classList.add("vertical-wrap");
+        }
     }
 
     function getDefaultTabIndex(folderId) {
-        switch (userSettings.get("landing-" + folderId)) {
-            case "guide":
-                return 1;
-            default:
-                return 0
+        if (userSettings.get("landing-" + folderId) === "guide") {
+            return 1;
         }
+        return 0;
     }
-    return function(view, params) {
+
+    return function (view, params) {
         function enableFullRender() {
-            return (new Date).getTime() - lastFullRender > 3e5
+            return new Date().getTime() - lastFullRender > 3e5;
         }
 
-        function onBeforeTabChange(e) {
-            preLoadTab(view, parseInt(e.detail.selectedTabIndex))
+        function onBeforeTabChange(evt) {
+            preLoadTab(view, parseInt(evt.detail.selectedTabIndex));
         }
 
-        function onTabChange(e) {
-            var previousTabController = tabControllers[parseInt(e.detail.previousIndex)];
-            previousTabController && previousTabController.onHide && previousTabController.onHide(), loadTab(view, parseInt(e.detail.selectedTabIndex))
+        function onTabChange(evt) {
+            var previousTabController = tabControllers[parseInt(evt.detail.previousIndex)];
+
+            if (previousTabController && previousTabController.onHide) {
+                previousTabController.onHide();
+            }
+
+            loadTab(view, parseInt(evt.detail.selectedTabIndex));
         }
 
         function getTabContainers() {
-            return view.querySelectorAll(".pageTabContent")
+            return view.querySelectorAll(".pageTabContent");
         }
 
         function initTabs() {
-            mainTabsManager.setTabs(view, currentTabIndex, getTabs, getTabContainers, onBeforeTabChange, onTabChange)
+            mainTabsManager.setTabs(view, currentTabIndex, getTabs, getTabContainers, onBeforeTabChange, onTabChange);
         }
 
         function getTabController(page, index, callback) {
             var depends = [];
+
+            // TODO int is a little hard to read
             switch (index) {
```

Comment:
this `switch` should probably be a lookup on some `map`-like thing instead

Reply:
TODO is already in place. As I said in the other PR, I think it requires a little more restructuring, so we avoid all these indices everywhere.